From 3e1b73371fa5a467fcd464d62b4c62499ac63f45 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Tue, 23 Aug 2016 02:05:53 -0700
Subject: [PATCH 20/36] UBUNTU: SAUCE: apparmor: deleted dentries can be
 disconnected

BugLink: http://bugs.launchpad.net/bugs/1615892

Signed-off-by: John Johansen <john.johansen@canonical.com>
Acked-by: Tim Gardner <tim.gardner@canonical.com>
Signed-off-by: Kamal Mostafa <kamal@canonical.com>
(cherry picked from commit 5de0db7215fc2f6979ecab26b6902d52dda28406)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 security/apparmor/path.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/security/apparmor/path.c b/security/apparmor/path.c
index 6cea2a6a1514..4d4549384bce 100644
--- a/security/apparmor/path.c
+++ b/security/apparmor/path.c
@@ -152,6 +152,9 @@ static int d_namespace_path(struct path *path, char *buf, char **name,
 
 	*name = res;
 
+	if (!connected)
+		error = disconnect(path, buf, name, flags, disconnected);
+
 	/* Handle two cases:
 	 * 1. A deleted dentry && profile is not allowing mediation of deleted
 	 * 2. On some filesystems, newly allocated dentries appear to the
@@ -164,9 +167,6 @@ static int d_namespace_path(struct path *path, char *buf, char **name,
 			goto out;
 	}
 
-	if (!connected)
-		error = disconnect(path, buf, name, flags, disconnected);
-
 out:
 	/*
 	 * Append "/" to the pathname.  The root directory is a special
-- 
2.17.1

